sympy/core/numbers.py

```diff
@@ -967,6 +967,8 @@ def __new__(cls, num, dps=None, prec=None, precision=None):
             num = '+inf'
         elif num is S.NegativeInfinity:
             num = '-inf'
+        elif type(num).__module__ == 'numpy': # support for numpy datatypes
+            num = sympify(num)
```

Comment:
Since you've factored out the code into a function you might as well call `convert_numpy_types` here instead of `sympify`.